### DISCUSSION TITLE
How to avoid building duckdb code to add extension support for new database attach?

### DISCUSSION THREAD

**Discussion by dawsondong** [duckdb/duckdb#14009] [Q&A] [answered]
If I want to write an extension of Oracle for DuckDB, and it support "attach" to mount the Oracle database, and then using regular SQL statements to access the Oracle database, does that mean I have to modify the mainline code of DuckDB? Because considering that in duckdb/main/extension_helper.hpp, the extension alias is defined: static const ExtensionAlias internal_aliases[] = {{"http", "httpfs"}, // httpfs
                                                  {"https", "httpfs"},
                                                  {"md", "motherduck"},       // motherduck
                                                  {"mysql", "mysql_scanner"}, // mysql
                                                  {"s3", "httpfs"},
                                                  {"postgres", "postgres_scanner"}, // postgres
                                                  {"sqlite", "sqlite_scanner"},     // sqlite
                                                  {"sqlite3", "sqlite_scanner"},
                                                  {nullptr, nullptr}};
I have to add one for Oracle extension, and then such statement could work: ATTACH 'dbname=xxx user=xxx host=127.0.0.1' AS db (TYPE ORACLE, READ_ONLY);
Does that mean I have to modify the code here and recompile the code for duckdb to have any chance of supporting my Oracle extension?

**Comment by carlopi** [comment]
For a demo/workaround, I would try can try using this syntax that should be already available:
```
D ATTACH 'extension_name:a b c d' (READ_ONLY);
IO Error: Extension "/Users/carlo/.duckdb/extensions/v1.1.0/osx_arm64/extension_name.duckdb_extension" not found.
```
I am not sure this works end-to-end, but might be worth trying.

This is something that we can consider expanding on, allowing extensions to register new mappings.

**Comment by carlopi** [comment]
Also, consider submitting this as a community extensions: https://community-extensions.duckdb.org/index.html

Sounds like something generically useful

**Comment by Tishj** [comment, marked as answer]
I think this is already possible, see <https://github.com/duckdb/postgres_scanner> which performs similar logic.
Having your extension register a StorageExtension that is identified as `ORACLE` should give you control over this in the registered attach function

**Reply by dawsondong** [reply]
Thanks. It did work.

**Comment by zhbdesign** [comment]
Has the Oracle plugin been completed? Can I try it out
